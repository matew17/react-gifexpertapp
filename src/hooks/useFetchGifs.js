import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(category).then((images) => {
            setstate({
                data: images,
                loading: false
            });
        });

    }, [category])

    return state;
}
