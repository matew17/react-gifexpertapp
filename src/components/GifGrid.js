import React from 'react'
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

function GifGrid({ category }) {
    const { data: images, loading } = useFetchGifs(category); // data:images renombre lo que llega como data por images

    return (
        <>
            <h3>{category}</h3>

            {loading && <p className="animate__animated animate__flash">Cargando</p>}

            <div className="card-grid">
                {
                    images && images.map(image => (

                        <GifGridItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
