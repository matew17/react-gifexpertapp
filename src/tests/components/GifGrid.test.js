import { shallow } from "enzyme";
import React from "react";
import '@testing-library/jest-dom';

import GifGrid from './../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('GifGrid', () => {

    const category = 'Kimetsu';
    // let wrapper = shallow(
    //     <GifGrid category={category} />
    // );

    // beforeEach(() => {
    //     wrapper = shallow(
    //         <GifGrid category={category} />
    //     );
    // });

    test('debe ser igual al snapshot ', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(
            <GifGrid category={category} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar imagenes cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'abc',
            url: 'https://url.com/image',
            title: 'kimetsu'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(
            <GifGrid category={category} />
        );

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
