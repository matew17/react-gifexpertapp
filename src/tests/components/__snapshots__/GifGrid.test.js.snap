// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid debe ser igual al snapshot  1`] = `
<Fragment>
  <h3>
    Kimetsu
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
